format/express-bunyan-logger.js

```diff
@@ -0,0 +1,26 @@
+const { fromJS } = require('immutable')
+
+const parseBunyanRequest = (strSource) => {
+  const source = typeof strSource === 'string' ? JSON.parse(strSource) : strSource
```

Comment:
Maybe we should assume that the JSON is already decoded. Like what we're doing in other formats. That might be the responsibility of the input to decode the JSON.

Reply:
Hmmm.
I believe we might run into issue doing this.
Typically I had my bunyan-logger writing to a file.
It would not make sense for the file input to try to decode as JSON, as it does not know about the format of the input.
Though if we assume the decode should be made by the input, I then cannot use my express-bunyan-logger format with the file input.

Reply:
I'm not that familiar with Bunyan but it seems that it is a JSON logging library so I wonder if Francois is right and we can assume that everything logged by bunyan is going to be JSON. In this case, I would parse the JSON in the input and have the formats convert the the parsed log object into our log format.

Reply:
Again I think it is a dangerous idea to do so as it makes the input expect a certain type of data from the source, which is I think not something the input should take care of.
My first example was with the file format. If we want the input to take care of the decoding, then we would need an input file-json, which is basically doing exactly the same thing as the current file input + decoding JSON, I don't find it really DRY.
The case for bunyan-logger + file can be found here : https://github.com/trentm/node-bunyan#stream-type-file
A second case (not exactly sure if it is a right case though, it's possible that in this case bunyan would not send logs in a JSON format) would be with this https://github.com/mcavage/node-bunyan-syslog
If bunyan is sending log as JSON to syslog, it will be sent as a stringified JSON, so again it would make more sense to ask the format, not the input, to take care of the parsing to JSON.
So we might assume everything is JSON, but it might be stringified JSON and I believe that our format should be the one taking the responsibility to decode if they need to.

Reply:
Ok I think you're right @jduthon. I thought Bunyan was an input but it looks like it is just a JSON-based format. So let's take care of the JSON parsing in the format.